Source/Core/Core/HW/SI_DeviceGBA.cpp

```diff
@@ -8,18 +8,75 @@
 #include "Common/StdMakeUnique.h"
 #include "Common/Thread.h"
 #include "Common/Logging/Log.h"
+#include "Core/CoreTiming.h"
 #include "Core/HW/SI_Device.h"
 #include "Core/HW/SI_DeviceGBA.h"
+#include "Core/HW/SystemTimers.h"
+#include "Core/HW/VideoInterface.h"
 
 #include "SFML/Network.hpp"
 
 static std::thread connectionThread;
+static std::thread clockThread;
 static std::queue<std::unique_ptr<sf::TcpSocket>> waiting_socks;
+static std::queue<std::unique_ptr<sf::TcpSocket>> waiting_clocks;
 static std::mutex cs_gba;
-namespace { volatile bool server_running; }
+static std::mutex cs_gba_clk;
+static u8 num_connected;
+
+namespace { volatile bool server_running; volatile bool clock_server_running; }
+
+enum EJoybusCmds
+{
+	CMD_RESET  = 0xff,
+	CMD_STATUS = 0x00,
+	CMD_READ   = 0x14,
+	CMD_WRITE  = 0x15
+};
+
+const u64 BITS_PER_SECOND = 115200;
```

Comment:
As discussed on IRC, the actual bps of the JOY link is closer to 220kbps, not 115.2kbps.